opennlp-tools/src/test/java/opennlp/tools/eval/AbstractEvalTest.java

```diff
@@ -18,73 +18,65 @@
 package opennlp.tools.eval;
 
 import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
 import java.math.BigInteger;
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.nio.charset.StandardCharsets;
 import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
 
 import org.junit.Assert;
 
 import opennlp.tools.ml.maxent.quasinewton.QNTrainer;
 import opennlp.tools.ml.naivebayes.NaiveBayesTrainer;
 import opennlp.tools.ml.perceptron.PerceptronTrainer;
+import opennlp.tools.util.ObjectStream;
 import opennlp.tools.util.TrainingParameters;
 import opennlp.tools.util.model.ModelUtil;
 
-public class EvalUtil {
+public abstract class AbstractEvalTest {
 
-  static final double ACCURACY_DELTA = 0.0001d;
+  public static final double ACCURACY_DELTA = 0.0001d;
+  public static final String HASH_ALGORITHM = "MD5";
 
-  static TrainingParameters createPerceptronParams() {
+  public static void verifyTrainingData(ObjectStream<?> samples, BigInteger checksum) throws Exception {
```

Comment:
For certain data sets it also takes quite some time to parse it once, reading the bytes from a file is much faster.